hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/CryptoInputStream.java

```diff
@@ -363,7 +363,23 @@ public int read(long position, final ByteBuffer buf)
 
     return n;
   }
-  
+
+  @Override
+  public void readFully(long position, ByteBuffer buf) throws IOException {
+    checkStream();
+    if (!(in instanceof ByteBufferPositionedReadable)) {
+      throw new UnsupportedOperationException("This stream does not support " +
+          "positioned reads with byte buffers.");
```

Comment:
The exception message could be confusing; it's not the CryptoInputStream that doesn't support byte buffer positioned reads, it's the input stream wrapped by it. How about printing the class name of the object in?

Reply:
Done.